src/taskpane/server.js

```diff
@@ -1176,6 +1283,11 @@ function protectColumn(sheet, columnNumber, rowLength, bgColor, name, hide) {
     // set the background color
     range.set({ format: { fill: { color: bgColor } } });
 
+    //hide the column, if necessary

```

Comment:
1. This function should be refactored as already mentioned
2. That will remove the hide param and the hide code from here.
3. NOTE: your code to hide the column is not working because I think you are using the wrong API call. it should be rangeOfColumns.hidden = true - see here: https://docs.microsoft.com/en-us/office/vba/api/excel.range.hidden

Reply:
I fixed the Hide function using range.columnHidden = true;